### PR TITLE
[CoSimulationApplication] Adding `AcuSolve` wrapper

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR introduces a new CoSimulation wrapper called `acuSolveWrapper` that serves as a dedicated Kratos wrapper for the [`acuSolve`](https://www.altair.com.es/Product/AcuSolve/) solver. 

**NOTE:** This wrapper class was not developed by me, the details I can provide are limited.

The contents of the commit include the following changes:

1. Definition of the `acuSolveWrapper` class that extends `CoSimulationSolverWrapper`.
2. Creation of model parts and associated variables within the Kratos environment using coupling data settings.
3. Reading arguments from a JSON file, including application name, working directory, problem directory, region, echo level, input file, number of processors, number of threads, GPU usage, restart flags, and time increment.
4. Launching the acuSolve solver based on the provided arguments using platform-specific commands.
5. Definition of an `Initialize` function for the `acuSolveWrapper` class to perform additional initialization tasks, such as importing coupling interfaces and configuring VTK output settings.
6. Importing of coupling interfaces based on model part names specified in settings.
7. Configuring and initializing VTK output for visualization purposes.

**🆕 Changelog**

- [Adding `AcuSolve` wrapper](https://github.com/KratosMultiphysics/Kratos/commit/54a061ed859c908d6281b158619c5cac19c47dbb)

### CODE REVIEW

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            if (gpu!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -gpu "+gpu+" -verbose "+str(echo_level)+" -lbuff"         
+            elif (rst!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -rst "+" -verbose "+str(echo_level)+" -lbuff"         
+            elif (frst!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -frst "+" -verbose "+str(echo_level)+" -lbuff"        
+            else:
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -verbose "+str(echo_level)+" -lbuff"  
+            cs_tools.cs_print_info(self.name + ": " +  cmd)
+            subprocess.Popen(cmd)
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: subprocess.run is the newer and safer alternative

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        cs_tools.cs_print_info(self.name + ": " +  "WRITING VTK OUTPUT...........")
+        vtk_output_configuration = KM.Parameters("""{
+                "model_part_name"        : \""""+model_part_name+"""\",
+                "output_sub_model_parts" : false,
+                "nodal_solution_step_data_variables" : ["HEAT_FLUX"]
+            }""")
+        self.vtk_output = VtkOutputProcess(self.model, vtk_output_configuration)
+        self.vtk_output.ExecuteInitialize()
+        self.vtk_output.ExecuteBeforeSolutionLoop()
+        self.vtk_output.PrintOutput()
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: Unconditional?

**Review by philbucher — DISMISSED** [review]
I recommend adding a test with a dummy solver, like I did in other tests

Otherwise this is entirely untested

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py (hunk 1 above)
Comment (loumalouomega) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: Okay

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py (hunk 2 above)
Comment (loumalouomega) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: Implementation is not mine...

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py (hunk 1 above)
Comment (loumalouomega) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: It was unused, replaced with os.system (common for Linux/windows)

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                cmd = base_cmd + common_cmd + " -rst " + verbose_cmd + buffer_cmd   
+            elif frst != "FALSE":
+                cmd = base_cmd + common_cmd + " -frst " + verbose_cmd + buffer_cmd
+            else:
+                cmd = base_cmd + common_cmd + verbose_cmd + buffer_cmd
+        else:
+            # Python exception in case not detected OS
+            raise Exception("Unsupported operating system detected.")
+        cs_tools.cs_print_info(self.name + ": " +  cmd)
+        os.system(cmd)
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py]: That is even worse 😅 (talking mainly about security issues, had sth similar at work)

I recommend using subprocess.run

**Comment by loumalouomega** [comment]
Then it will be... Tomorrow

**Review by philbucher — APPROVED** [review]
Nice

**Pull request merged**